packages/orbit-components/src/Tabs/components/TabList/index.tsx

```diff
@@ -18,6 +18,50 @@ const TabList = ({
   ariaLabel,
   ariaLabelledby,
 }: Props) => {
+  const tabListRef = React.useRef<HTMLDivElement>(null);
+
+  const handleKeyDown = React.useCallback((event: React.KeyboardEvent<HTMLDivElement>) => {
+    // Only handle keyboard navigation if we have a reference to the tab list
+    if (!tabListRef.current) return;
+
+    // Get all non-disabled tab elements
+    const tabs = Array.from(
+      tabListRef.current.querySelectorAll('button[role="tab"]:not([disabled])'),
+    );
+
+    if (tabs.length === 0) return;
+
+    const currentIndex = tabs.findIndex(tab => tab === document.activeElement);
+    let newIndex: number | null = null;
+
+    // Handle keyboard navigation
```

Comment:
We're executing all the logic above regardless of the key pressed. Would it make sense to add a check to avoid running it? Something like this:

```tsx
// Early return if it's not a key we handle - avoid expensive operations
if (!["ArrowRight", "ArrowLeft", "Home", "End"].includes(event.key)) {
  return;
}
```

Reply:
Good point. Added

Reply:
Did you push this? 🦮